src/runtime/graph/graph_runtime.cc

```diff
@@ -56,6 +57,25 @@ class GraphRuntime : public ModuleNode {
       if (op_execs_[i]) op_execs_[i]();
     }
   }
+
+  void DebugRun() {
+    struct timeval tp;
+    // Execute each op and copy the outs
+    for (size_t i = 0; i < op_execs_.size(); ++i) {
+      if (op_execs_[i]) op_execs_[i]();
+      gettimeofday(&tp, NULL);
+      int64_t mstime = int64_t(tp.tv_sec * 1000000L + tp.tv_usec);
+      size_t num_outputs = (nodes_[i].op_type == "null") ? 1: nodes_[i].param.num_outputs;
+      for (size_t j = 0; j < num_outputs; j++) {
+          uint32_t eid = this->entry_id(i, j);
+          TVM_CCALL(TVMArrayCopyFromTo(&data_entry_[eid],
+                                       &debug_buffers_[eid]->out_tensor,
+                                       nullptr));
+          debug_buffers_[eid]->time_stamp = mstime;
```

Comment:
I prefer to use layer-wise time cost to replace timestamp here.
To say, set timer to only record the time cost of 
`if (op_execs_[i]) op_execs_[i]();`

The way you record time stamp also includes the overhead of data copy.  These time stamps are not identical to the real time stamps when debug mode is off.